### PR TITLE
8.3 update to 1.1.1k-9

### PR DESCRIPTION
**Pull request by bleader**
- Update to 1.1.1k-9 from CentOS 8 Stream
- Renamed Patch100 to Patch500 as original package now has Patch100
- Removed Patch1000 (update-expired-SCT-certificates.backport) as it is now integrated upstream

### CODE REVIEW

**Review by benjamreis — APPROVED** [review]

File: SPECS/xs-openssl.spec [hunk 1]
```diff
@@ -84,13 +84,20 @@ Patch56: openssl-1.1.1-s390x-ecc.patch
...
+Patch86: openssl-1.1.1-cve-2022-2097.patch
+#OpenSSL 1.1.1t CVEs
+Patch101: openssl-1.1.1-cve-2022-4304-RSA-oracle.patch
+Patch102: openssl-1.1.1-cve-2022-4450-PEM-bio.patch
+Patch103: openssl-1.1.1-cve-2023-0215-BIO-UAF.patch
+Patch104: openssl-1.1.1-cve-2023-0286-X400.patch
 
 
-Patch100: use-linux-random-header.patch
+Patch500: use-linux-random-header.patch
```
Comment (stormi) [inline SPECS/xs-openssl.spec:100]: Let's document where this patch comes from
```suggestion
# XenServer patch
Patch500: use-linux-random-header.patch
```

File: SPECS/xs-openssl.spec [hunk 2]
```diff
@@ -524,6 +538,34 @@ export LD_LIBRARY_PATH
...
+
+- * Tue Jul 05 2022 Clemens Lang <cllang@redhat.com> - 1:1.1.1k-7
+- - Fix CVE-2022-2097: AES OCB fails to encrypt some bytes on 32-bit x86
+    Resolves: CVE-2022-2097
+- - Update expired certificates used in the testsuite
+    Resolves: rhbz#2092462
+- - Fix CVE-2022-1292: openssl: c_rehash script allows command injection
+    Resolves: rhbz#2090372
+- - Fix CVE-2022-2068: the c_rehash script allows command injection
+    Resolves: rhbz#2098279
```
Comment (stormi) [inline SPECS/xs-openssl.spec]: If I were to re-establish the changelog embedding syntax, I'd probably do it differently, without all these dashes, but for now the convention is

```suggestion
- *** Upstream changelog ***
- * Wed Feb 08 2023 Dmitry Belyavskiy <dbelyavs@redhat.com> - 1:1.1.1k-9
- - Fixed Timing Oracle in RSA Decryption
-   Resolves: CVE-2022-4304
- - Fixed Double free after calling PEM_read_bio_ex
-   Resolves: CVE-2022-4450
- - Fixed Use-after-free following BIO_new_NDEF
-   Resolves: CVE-2023-0215
- - Fixed X.400 address type confusion in X.509 GeneralName
-   Resolves: CVE-2023-0286
- * Thu Jul 21 2022 Dmitry Belyavskiy <dbelyavs@redhat.com> - 1:1.1.1k-8
- - Fix no-ec build
-   Resolves: rhbz#2071020
- * Tue Jul 05 2022 Clemens Lang <cllang@redhat.com> - 1:1.1.1k-7
- - Fix CVE-2022-2097: AES OCB fails to encrypt some bytes on 32-bit x86
-   Resolves: CVE-2022-2097
- - Update expired certificates used in the testsuite
-   Resolves: rhbz#2092462
- - Fix CVE-2022-1292: openssl: c_rehash script allows command injection
-   Resolves: rhbz#2090372
- - Fix CVE-2022-2068: the c_rehash script allows command injection
-   Resolves: rhbz#2098279
```

File: SPECS/xs-openssl.spec [hunk 3]
```diff
@@ -532,7 +574,7 @@ export LD_LIBRARY_PATH
 - *** Upstream changelog ***
 - * Wed Mar 23 2022 Clemens Lang <cllang@redhat.com> - 1:1.1.1k-6
 - - Fixes CVE-2022-0778 openssl: Infinite loop in BN_mod_sqrt() reachable when parsing certificates
-- - Resolves: rhbz#2067144
+- - Resolves: rhbz#2067146
```
Comment (stormi) [inline SPECS/xs-openssl.spec:576]: Any idea why this changed? Was the previous reference wrong? Same question for the other similar change.

Comment (bleader) [reply SPECS/xs-openssl.spec:576]: Not really, they do change them in a lot of update. I would assume `rhbz` stands for redhat bugzilla (which as far as I can tell we do not have access to), and they have an internal process where they clone tickets for further version, and somehow update the references to newer tickets for newer versions. I was wondering if we wanted to integrate these changes or not. So I actually integrated them so we can decide this together.

Comment (bleader) [reply SPECS/xs-openssl.spec:576]: I forgot to add that I personally would not integrate them, less changes in the diff, clearer patch, but maybe more diff when updating further.

Comment (stormi) [reply SPECS/xs-openssl.spec:576]: Let's keep as you did, synced with the upstream specfile. I prefer a larger diff now over a larger diff later.

File: SPECS/xs-openssl.spec (hunk 1 above)
Comment (bleader) [reply SPECS/xs-openssl.spec:100]: Done.

File: SPECS/xs-openssl.spec (hunk 2 above)
Comment (bleader) [reply SPECS/xs-openssl.spec]: Thanks, I was unsure about it, fixed.

**Comment by bleader** [comment]
Updated by re-doing the upstream import using `xcp/scripts/import_srpm.py` which was a good thing as my original import was using the git repo's c8s branch and wasn't actually matching `1.1.1k-9`. We had one more patch (not present in the specfile, but in the SOURCES/ folder).

**Review by stormi — APPROVED** [review]

**Comment by bleader** [comment]
Branch `upstream` was pushed now, in order to be able to merge this when ready for packaging and tests.

**Comment by stormi** [comment]
Koji build (v8.3-incoming): http://koji.xcp-ng.org/taskinfo?taskID=39359

**Pull request merged**